curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/645b9ea45d3a9a9d711df81a.md

```diff
@@ -0,0 +1,172 @@
+---
+id: 645b9ea45d3a9a9d711df81a
+title: Step 17
+challengeType: 0
+dashedName: step-17
+---
+
+# --description--
+
+Finally, you should clear the number input by setting its value to an empty string. Then later when you convert several numbers in a row, you won't have to delete the previous number before entering the next one.
```

Comment:
Awesome. Thanks for your kind words, @gikf. That means a lot coming from a very detailed reviewer like yourself. Fortunately I had a great prototype and notes from @jdwilkin4 to work with, so I mostly just converted those into steps.

Also, thanks for that helpful suggestion. I like the idea of displaying the original decimal number so learners can double check the conversion. The easiest thing now would be to change or remove the step where the input gets cleared, but it would be nicer overall to clear the input and display the decimal number somewhere on the page.

I'll chat with the rest of the JS curriculum team about our options and see what they think.

Reply:
Note: I'll work updating the UI for this project to show the original decimal number in a followup PR.